app/services/preserved_object_handler.rb

```diff
@@ -191,8 +191,9 @@ def confirm_version_in_catalog
     results = []
     confirm_results = with_active_record_transaction_and_rescue do
       po_db_object = PreservedObject.find_by!(druid: druid)
+      pc_db_object = PreservedCopy.find_by!(preserved_object: po_db_object, endpoint: endpoint) if po_db_object
+      break unless pc_db_object
```

Comment:
I like this very much. Should we have a test for this?

Reply:
good idea - I just added a commit with one.